R/shinyGadget.R

```diff
@@ -1,29 +1,98 @@
-basicProjectGadget <- function() {
-
-ui <- miniUI::miniPage(
-  miniUI::gadgetTitleBar("Create Basic pRoject"),
-  miniUI::miniContentPanel(
-    shiny::inputPanel(
-      shiny::textInput("name", "Project Name")
-    ),
-    shiny::inputPanel(
-      shiny::checkboxInput("readme", "Create README", value = TRUE),
-      shiny::checkboxInput("git", "Use Git", value = TRUE),
-      shiny::checkboxInput("travis", "Use Travis", value = TRUE),
-      shiny::checkboxInput("packrat", "Use Packrat", value = TRUE)
+projectGadget <- function() {
+
+  ui <- miniUI::miniPage(
+    miniUI::gadgetTitleBar("Create pRoject"),
+    miniUI::miniContentPanel(
+      shiny::inputPanel(
+        shiny::selectInput(
+          "projectType", label = "Project Type",
+          choices = c("Basic","Analysis","Package","Training"),
+          selected = "Basic"
+        ),
+        shiny::textInput("name", "Project Name")
+      ),
+      shiny::uiOutput("ui"),
+      shiny::conditionalPanel(
+        condition = "input.projectType != 'Package'",
+        shiny::inputPanel(
+          shiny::h4("Best Practices"),
+          shiny::br(),
+          shiny::checkboxInput("readme", "Create README", value = TRUE),
+          shiny::checkboxInput("git", "Use Git", value = TRUE),
+          shiny::checkboxInput("travis", "Use Travis", value = TRUE),
+          shiny::checkboxInput("packrat", "Use Packrat", value = TRUE)
+        )
+      )
     )
   )
-)
 
-server <- function(input, output, session) {
+  server <- function(input, output, session) {
 
-  shiny::observeEvent(input$done, {
-    createBasicProject(name = input$name, readme = input$readme,
-                       git = input$git, travis = input$travis,
-                       packrat = input$packrat)
-    shiny::stopApp()
-  })
-}
+    output$ui <- renderUI({
+
+      switch(input$projectType,
+
+        "Basic"     = shiny::div(),
+        "Analysis"  = shiny::inputPanel(
+                        shiny::selectInput("dirs", label = "Directories",
```

Comment:
@jonmcalder Do you think people will object to the defaults and/or prefer their own such that this should be some sort of free text field?